test/test_decorator.py

```diff
@@ -69,29 +75,41 @@ def _wrapper(*other_args, **other_kwargs):
     return _wrapper
 
 
+def _xfail_in_py311(*param):
+    return pytest.param(
+        *param,
+        marks=pytest.mark.xfail(
+            sys.version_info >= (3, 11),
+            reason="TODO: test doesn't work in Python 3.11",
+            strict=True,
+        ),
+    )
+
+
 @pytest.mark.parametrize(
-    "args, expected, fn",
+    "args, expected, delay_wrapper",
     [
         ("", 1, partial(_fn_with_positional_only, 1)),
         ("2", 2, partial(_fn_with_positional_only, 1)),
-        ("2", 2, _fn_with_positional_only),
+        ("2", 2, partial(_fn_with_positional_only)),
```

Comment:
Hey there @zhiruiluo , Would you mind explaining a few things:
- why is the `delay_wrapper` is necessary here?
- Did the `inspect` module change from 3.11 to 3.11.3? Is that what broke this test?

Reply:
The recently PR https://github.com/python/cpython/pull/103557 disallows pos-or-kw params without default after pos-only with default, which breaks the test for Python 3.11.4 released on June 6.

Reply:
They will likely backport this change to all Python 3.11 verions.